Libraries/LibDebug/DebugInfo.h

```diff
@@ -45,6 +49,33 @@ class DebugInfo {
         bool operator!=(const SourcePosition& other) const { return !(*this == other); }
     };
 
+    struct VariableInfo {
+        enum class LocationType {
+            None,
+            Address,
+            Regsiter,
+        };
+        String name;
+        String type;
+        LocationType location_type { LocationType::None };
+        union {
+            u32 address;
+        } location_data { 0 };
+
+        NonnullOwnPtrVector<VariableInfo> members;
+        VariableInfo* parent { nullptr };
+    };
+
+    struct VariablesScope {
+        bool is_function { false };
+        Optional<String> name;
```

Comment:
`String` has a null state which is what you get with `String()`, we normally use that instead of wrapping it in `Optional`.